compiler/wasm/src/compile_new.rs

```diff
@@ -0,0 +1,258 @@
+use crate::compile::{
+    file_manager_with_source_map, preprocess_contract, preprocess_program, JsCompileResult,
+    PathToFileSourceMap,
+};
+use crate::errors::{CompileError, JsCompileError};
+use noirc_driver::{
+    add_dep, compile_contract, compile_main, prepare_crate, prepare_dependency, CompileOptions,
+};
+use noirc_frontend::{
+    graph::{CrateGraph, CrateId, CrateName},
+    hir::Context,
+};
+use std::path::Path;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[wasm_bindgen]
+pub struct ContextWrapper {
+    context: Context,
+}
+
+#[wasm_bindgen]
+#[derive(Debug, Copy, Clone)]
+pub struct CrateIDWrapper(CrateId);
```

Comment:
We can have a conditional config for the wasm_bindgen attribute which would allow us to get rid of this wrapper struct entirely

Reply:
That would require us to make the compiler crates aware of wasm-bindgen though. I'd say this wrapper is fine in order to avoid having to do that.

Reply:
I think thats okay given it is conditionally compiled in anyways -- thats what features are for really -- I'm not strongly for this strategy so I'm happy to table it